ui/app/modifiers/code-mirror.js

```diff
@@ -0,0 +1,60 @@
+import { action } from '@ember/object';
+import { bind } from '@ember/runloop';
+import codemirror from 'codemirror';
+import Modifier from 'ember-modifier';
+
+import 'codemirror/addon/edit/matchbrackets';
+import 'codemirror/addon/selection/active-line';
+import 'codemirror/addon/lint/lint.js';
+import 'codemirror/addon/lint/json-lint.js';
+// right now we only use the ruby and javascript, if you use another mode you'll need to import it.
+// https://codemirror.net/mode/
+import 'codemirror/mode/ruby/ruby';
+import 'codemirror/mode/javascript/javascript';
```

Comment:
Given that this isn't actually Javascript either, but JSON, is there a JSON-specific mode so that we don't carry along anything needed for Javascript editing too? (Which I imagine is non-trivial? Not sure how this works under the hood)

Reply:
I gave this a shot and it turns out that codemirror's javascript mode/parser is actually really good at color-highlighting HCL!

![image](https://user-images.githubusercontent.com/713991/176459755-db1fd0fd-de7f-4e82-8ffe-72544ff1294c.png)

it's what IvyCodemirror was using behind the scenes. So while the file is non-trivially large, the language mode file should still be included.

Dropping Ruby, though.